### PR TITLE
Room owner

### PR DESCRIPTION
**Pull request by Crysicia**
Adds:
- Room owner can change room password of remove it
- Room owner can make other chatroom users admin or demote them from admin
- Refactor chatroom service

### CODE REVIEW

File: frontend/src/components/EditChatroomDialog.vue
Comment (tsiguenz) [inline frontend/src/components/EditChatroomDialog.vue:1]: add delete chatrooms :pleading_face:

**Review by tsiguenz — COMMENTED** [review]
This looks good!
When the owner leave his room, the room no belongs to anyone. I think we can own a room without being in it or change the owner by the  first admin or something like that.

File: backend/src/chatroom_user/chatroom_user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  })
+  @ApiParam({
+    name: 'id',
+    type: String,
+    required: true,
+    description: 'User id'
+  })
+  @Post(':id/demote')
+  async demote(@User() user: object, @Param('chatroomId') chatroomId: string, @Param('id') userId: string) {
+    if (!(await this.chatroomUserService.isUserOwner(user['id'], chatroomId)) || !(await this.chatroomUserService.isUserAdmin(userId, chatroomId))) {
```
[the remark this answers is not in the document]
Comment (Crysicia) [reply backend/src/chatroom_user/chatroom_user.controller.ts]: No, it checks if the user making the request is Owner and if the user getting demoted is admin

File: frontend/src/components/EditChatroomDialog.vue [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+<script>
+import axios from 'axios';
+import * as constants from '@/constants';
+import swal from 'sweetalert';
+import formatError from '@/utils/lib';
+import ChatService from '../services/chat.service';
+
+export default {
+  emits: ['create', 'delete'],
```
Comment (gmorange42) [inline frontend/src/components/EditChatroomDialog.vue:47]: Remove create

**Review by gmorange42 — COMMENTED** [review]
GG

**Pull request merged**